### PR TITLE
New: Add default severity to reports hint-meta-charset-utf-8

### LINKED ISSUE #3231: [Severity Updates] hint-meta-charset-utf-8
| Report | Severity |
| --------| ----------|
| Report message when the 'charset' meta element is duplicated: 'charset' meta element is not needed as one was already specified. | Warning |
| Report message when the 'charset' meta element is not the first element in '<head>': 'charset' meta element should be the first thing in '<head>'. | Hint if within 1024, Error if after 1024 |
| Report message when the 'charset' meta element is in the '<body>': 'charset' meta element should be specified in the '<head>', not '<body>'. | Error |
| Report message when the 'charset' meta element was not specified: 'charset' meta element was not specified. | Warning |
| Report message when the 'charset' meta element is used in a long form: 'charset' meta element should be specified using shorter '<meta charset="utf-8">' form. | Warning |
| Report message when the 'charset' meta element has a wrong value: 'charset' meta element value should be 'utf-8', not '$1'. | Warning if "utf8", Error otherwise |

### PR DESCRIPTION
**Pull request by molant**
## Pull request checklist

Make sure you:

- [x] Signed the [Contributor License Agreement](https://cla.js.foundation/webhintio/hint)
- [x] Followed the [commit message guidelines](https://webhint.io/docs/contributor-guide/getting-started/pull-requests/#commit-messages)

For non-trivial changes, please make sure you also:

- ~~Added/Updated related documentation.~~
- [x] Added/Updated related tests.

## Short description of the change(s)

Fix #3231

### CODE REVIEW

File: packages/hint-meta-charset-utf-8/src/hint.ts [hunk 1]
```diff
@@ -98,15 +123,35 @@ export default class MetaCharsetUTF8Hint implements IHint {
             const isMetaElementFirstHeadContent = (/^<head[^>]*>\s*<meta/).test(headElementContent);
 
             if (!isCharsetMetaFirstHeadElement || !isMetaElementFirstHeadContent) {
-                context.report(resource, getMessage('metaElementFirstThing', context.language), { element: charsetMetaElement });
+
+                const index = html.indexOf(charsetMetaElement.outerHTML);
```
Comment (antross) [inline packages/hint-meta-charset-utf-8/src/hint.ts]: We actually have the start and end offset from `parse5`, but haven't plumbed it through yet. I think a minor update to `HTMLElement.getLocation` to expose the start and end offset would be better than using a serialization (which may fail to find matching text at all. Best to do after the flattening PR is merged of course.

Comment (molant) [reply packages/hint-meta-charset-utf-8/src/hint.ts]: Done, can you take a look? Had to split in a few commits

File: packages/hint-meta-charset-utf-8/src/hint.ts [hunk 2]
```diff
@@ -98,15 +124,32 @@ export default class MetaCharsetUTF8Hint implements IHint {
             const isMetaElementFirstHeadContent = (/^<head[^>]*>\s*<meta/).test(headElementContent);
 
             if (!isCharsetMetaFirstHeadElement || !isMetaElementFirstHeadContent) {
-                context.report(resource, getMessage('metaElementFirstThing', context.language), { element: charsetMetaElement });
+
+                const severity = firstHeadElement.getLocation().endOffset || 0 <= 1024 ?
```
Comment (antross) [inline packages/hint-meta-charset-utf-8/src/hint.ts]: Guess it's about type to upgrade to TypeScript 3.7. Ready for that nullish coalescing! 😉

Comment (molant) [reply packages/hint-meta-charset-utf-8/src/hint.ts]: Yes please 😉

https://twitter.com/molant/status/1193293388822401025?s=20

**Pull request merged**